src/main/java/com/mraof/minestuck/entity/ai/attack/FireballShootGoal.java

```diff
@@ -0,0 +1,129 @@
+package com.mraof.minestuck.entity.ai.attack;
+
+import com.mraof.minestuck.entity.AttackingAnimatedEntity;
+import com.mraof.minestuck.entity.animation.PhasedMobAnimation;
+import net.minecraft.core.BlockPos;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.projectile.SmallFireball;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Alternative attack for when the attack target is out of standard range, a small fireball is shot at the targets initial position.
+ */
+public class FireballShootGoal<T extends AttackingAnimatedEntity> extends AnimatedAttackWhenInRangeGoal<T>
+{
+	public static final int GROUP_SHOOT_COOLDOWN = 20; //how long it should take before another nearby mob can shoot fireballs
+	
+	/**
+	 * Block position that the target was standing at when start() was ran
+	 */
+	private BlockPos initialTargetPos;
+	
+	public FireballShootGoal(T entity, PhasedMobAnimation animation, float minRange, float maxRange, int actionCooldown)
+	{
+		super(entity, animation, minRange, maxRange, actionCooldown);
+	}
+	
+	@Override
+	public boolean canUse()
+	{
+		return super.canUse() && entity.hasFinishedCooldown() && noEntitiesObstructing(this.entity.getTarget()); //the super implies the target is not null
+	}
+	
+	@Override
+	public void start()
+	{
+		super.start();
+		
+		initialTargetPos = Objects.requireNonNull(this.entity.getTarget()).blockPosition();
+		
+		//code block handles group cooldown
```

Comment:
I recommend instead extracting the code block as a function named like that

Reply:
it hath been done